### PR TITLE
Add datastore command

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

To migrate data from MongoDB to MySQL, use the following command:
```bash
$ pipectl datastore migrate \
	--upstream-data-src=*mongodb-datastore-url* \
	--downstream-data-src=*mysql-datastore-url* \
	--database=*database-name*
```

**Which issue(s) this PR fixes**:

Ref #1792 

**Does this PR introduce a user-facing change?**:

```release-note
Add pipectl datastore command
```

### CODE REVIEW

File: pkg/datastore/migration/data_transfer.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	for _, kind := range kinds {
+		eg.Go(func() error {
+			return transferOne(ctx, d.source, d.destination, kind)
+		})
+	}
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (nakabonne) [inline pkg/datastore/migration/data_transfer.go:87]: nit
```suggestion
	return eg.Wait()
```

File: pkg/app/pipectl/cmd/datastore/migrate.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to connect to downstream datastore: %w", err)
+	}
+
+	modelsNameList, err := makeMigrateModelsList(m.models)
+	if err != nil {
+		return fmt.Errorf("failed to migrate datastore: %w", err)
+	}
+
+	if err = migration.NewDataTransfer(mongodbDatastore, mysqlDatastore).TransferMulti(ctx, modelsNameList); err != nil {
```
Comment (nghialv) [inline pkg/app/pipectl/cmd/datastore/migrate.go:79]: nit: Break this into multiple lines

```golang
transfer := NewDataTransfer(..)
transfer.TranferMulti()
```

File: pkg/app/pipectl/cmd/datastore/migrate.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		return fmt.Errorf("failed to migrate datastore: %w", err)
+	}
+
+	return nil
+}
+
+func makeMigrateModelsList(modelsName []string) ([]string, error) {
```
Comment (nghialv) [inline pkg/app/pipectl/cmd/datastore/migrate.go:86]: nit: `makeMigrateModelList`

File: pkg/datastore/datastore.go [hunk 4]
```diff
@@ -38,6 +38,18 @@ var (
 	ErrUnimplemented   = errors.New("unimplemented")
 )
 
+// MigratableModelsKind is a slide of models' name which available to migrate.
+var MigratableModelsKind = []string{
```
Comment (nghialv) [inline pkg/datastore/datastore.go:42]: nit: `MigratableModelKinds`

File: pkg/datastore/migration/data_transfer.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+type modelData interface {
+	GetId() string
+}
+
+func makeModelObject(kind string) (modelData, error) {
+	switch kind {
+	case "Project":
```
Comment (nghialv) [inline pkg/datastore/migration/data_transfer.go:96]: Make the kind defined in the datastore package to be public constants to use here is fine.

Comment (khanhtc1202) [reply pkg/datastore/migration/data_transfer.go:96]: Let's me address it by other PR since bunch of stores have to be updated as well 👀

Comment (nghialv) [reply pkg/datastore/migration/data_transfer.go:96]: Sure. 👍

**Comment by khanhtc1202** [comment]
@nakabonne @nghialv thanks for the comments 🙏 
addressed on https://github.com/pipe-cd/pipe/pull/1806/commits/7408b28a3471f3fc5ecf4147b21ece102208f52b PTAL 👀

**Comment by nakabonne** [comment]
Thank you
/lgtm

File: pkg/app/pipectl/cmd/datastore/migrate.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	modelsNameList, err := makeMigrateModelList(m.models)
+	if err != nil {
+		return fmt.Errorf("failed to migrate datastore: %w", err)
+	}
+
+	dataTransfer := migration.NewDataTransfer(mongodbDatastore, mysqlDatastore)
+	if err = dataTransfer.TransferMulti(ctx, modelsNameList); err != nil {
+		return fmt.Errorf("failed to migrate datastore: %w", err)
+	}
+
```
Comment (nghialv) [inline pkg/app/pipectl/cmd/datastore/migrate.go:83]: Add log to show that the migration process was completed succesfully.

Comment (khanhtc1202) [reply pkg/app/pipectl/cmd/datastore/migrate.go:83]: You're right 🤣 https://github.com/pipe-cd/pipe/pull/1806/commits/301503f5a22e215b295a9ae06a3fad9c4972f323 🙏

Comment (khanhtc1202) [reply pkg/app/pipectl/cmd/datastore/migrate.go:87]: thanks 🙏  addressed on https://github.com/pipe-cd/pipe/pull/1806/commits/ac9b7bbfea2d36cf43d7579af1e9c3fa0ed2b4f9 👀

**Comment by nghialv** [comment]
Thank you.
/approve

**Pull request merged**